projects/graphql-client/src/graphql-apollo-integration.test.ts

```diff
@@ -0,0 +1,243 @@
+
+    // Make second request
+    spectator.service
+      .query({
+        query: gql(buildRequestString()),
+        errorPolicy: 'all',
+        fetchPolicy: 'cache-first'
+      })
+      .subscribe();
+    spectator.controller.verify();
+    flush();
+  }));
+
+  test('should not cache on identical queries when no cache policy is selected', fakeAsync(() => {
+    spectator = createService();
+    spectator.service
+      .query({
+        query: gql(buildRequestString()),
+        errorPolicy: 'all',
+        fetchPolicy: 'cache-first'
+      })
+      .subscribe();
+    // Flush first request out, otherwise apollo deduping will always reuse an identical in flight query
+    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+
+    // Make second request
+    spectator.service
+      .query({
+        query: gql(buildRequestString()),
+        errorPolicy: 'all',
+        fetchPolicy: 'no-cache'
+      })
+      .subscribe();
+    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+    spectator.controller.verify();
+    flush();
+  }));
+
+  test('only fires queries that are subscribed', fakeAsync(() => {
+    spectator = createService();
+    const query$ = spectator.service.query({
+      query: gql(buildRequestString()),
+      errorPolicy: 'all',
+      fetchPolicy: 'cache-first'
+    });
+    tick();
+    spectator.controller.expectNone(graphQlUri);
+    query$.subscribe();
+
+    // Flush first request out, otherwise apollo deduping will always reuse an identical in flight query
+    spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+    flush();
+  }));
+
+  // test('does not cache entities, even if they share an id', fakeAsync(() => {
+  //   spectator.service.registerHandler(buildQueryHandler());
+  //   spectator.service.queryImmediately(buildRequest()).subscribe();
+  //   // Return an object to be cached
+  //   spectator.expectOne(graphQlUri, HttpMethod.POST).flush(buildServerResponse());
+  //   // Make a new query, but tweak params so it's not cached and return same id with differen val
+  //   const modifiedRequest = buildRequest();
+  //   modifiedRequest.select.arguments = [{ name: 'a', value: 'b' }];
+  //   spectator.service.queryImmediately(modifiedRequest).subscribe();
+  //   // Return a response with same id but different value
+  //   const modifiedResponse = buildServerResponse();
+  //   modifiedResponse.data.test.value = 'baz';
+  //   spectator.expectOne(graphQlUri, HttpMethod.POST).flush(modifiedResponse);
+  //   // Now make the first request again - it should be the original result
+  //   spectator.service.queryImmediately(buildRequest()).subscribe(result => {
+  //     expect((result as { value: string }).value).toEqual('bar');
+  //   });
+  //   // Tick, our responses are always async
+  //   tick();
+  // }));
+
+  // test('late subscribers to a debounced query can access results', fakeAsync(() => {
```

Comment:
This test fails now. Resubscribing do not fire a new query for observable returned by apollo. Do we need this behavior?